Modules/vc-module-orders/VirtoCommerce.OrdersModule.Data/Repositories/OrderRepository.cs

```diff
@@ -25,19 +26,21 @@ public OrderRepository(OrderDbContext dbContext, IUnitOfWork unitOfWork = null)
         public IQueryable<LineItemEntity> LineItems => DbContext.Set<LineItemEntity>();
         public IQueryable<PaymentGatewayTransactionEntity> Transactions => DbContext.Set<PaymentGatewayTransactionEntity>();
 
-        public virtual async Task<CustomerOrderEntity[]> GetCustomerOrdersByIdsAsync(string[] ids, CustomerOrderResponseGroup responseGroup)
+        public virtual async Task<CustomerOrderEntity[]> GetCustomerOrdersByIdsAsync(string[] ids, string responseGroup = null)
         {
+            var customerOrderResponseGroup = EnumUtility.SafeParseFlags(responseGroup, CustomerOrderResponseGroup.Full);
+
             var result = await CustomerOrders.Where(x => ids.Contains(x.Id)).ToArrayAsync();
             var orderDiscounts = Discounts.Where(x => ids.Contains(x.CustomerOrderId)).ToArrayAsync();
             var orderTaxDetails = TaxDetails.Where(x => ids.Contains(x.CustomerOrderId)).ToArrayAsync();
             await Task.WhenAll(orderDiscounts, orderTaxDetails);
 
-            if ((responseGroup & CustomerOrderResponseGroup.WithAddresses) == CustomerOrderResponseGroup.WithAddresses)
+            if ((customerOrderResponseGroup & CustomerOrderResponseGroup.WithAddresses) == CustomerOrderResponseGroup.WithAddresses)
```

Comment:
Replace please to `HasFlags` expression everywhere

Reply:
> And please move all ResponseGroup enum types into Data projects from Core for all modules.

No way!
It should be possible to call `ICustomerOrderService.GetByIdAsync("123", CustomerOrderResponseGroup.WithItems.ToString())` without adding Data project to dependencies.

Reply:
fixed